### PR TITLE
refactor: swap out NewDisplayCurrencyConverter

### PR DESCRIPTION
**Pull request by vindard**
## Description

This PR swaps out the `NewDisplayCurrencyConverter` for `DisplayPriceRatio` for conversions.

### CODE REVIEW

File: src/domain/shared/primitives.ts [hunk 1]
```diff
@@ -92,6 +92,19 @@ export const paymentAmountFromNumber = <T extends WalletCurrency>({
   }
 }
 
+export const displayAmountFromNumber = <T extends DisplayCurrency>({
+  amount,
+  currency,
+}: {
+  amount: number
+  currency: T
+}): DisplayAmount<T> | ValidationError => {
```
Comment (dolcalmi) [inline src/domain/shared/primitives.ts:101]: is `ValidationError` needed?

File: src/app/payments/send-lightning.ts [hunk 2]
```diff
@@ -322,11 +333,13 @@ const executePaymentViaIntraledger = async <
   paymentFlow,
   senderWallet,
   senderUsername,
+  senderDisplayCurrency,
   memo,
 }: {
   paymentFlow: PaymentFlow<S, R>
   senderWallet: WalletDescriptor<S>
   senderUsername: Username | undefined
+  senderDisplayCurrency: DisplayCurrency
```
Comment (dolcalmi) [inline src/app/payments/send-lightning.ts:342]: these two fields are from account.. maybe is better to pass `senderAccount` as param

File: src/app/payments/send-lightning.ts [hunk 3]
```diff
@@ -417,9 +429,9 @@ const executePaymentViaIntraledger = async <
           pubkey: recipientPubkey,
           paymentAmounts: paymentFlow,
 
-          amountDisplayCurrency: converter.fromUsdAmount(paymentFlow.usdPaymentAmount),
+          amountDisplayCurrency: amountDisplayCurrencyAsNumber,
           feeDisplayCurrency: 0 as DisplayCurrencyBaseAmount,
-          displayCurrency: DisplayCurrency.Usd,
+          displayCurrency: senderDisplayCurrency,
```
Comment (dolcalmi) [inline src/app/payments/send-lightning.ts:434]: same here, we need to pass `additionalCreditMetadata` (line 452)

File: src/app/wallets/send-on-chain.ts [hunk 4]
```diff
@@ -291,7 +302,7 @@ const executePaymentViaIntraledger = async <
           sendAll,
           paymentAmounts: paymentFlow,
 
-          amountDisplayCurrency: converter.fromUsdAmount(paymentFlow.usdPaymentAmount),
+          amountDisplayCurrency: amountDisplayCurrencyAsNumber,
```
Comment (dolcalmi) [inline src/app/wallets/send-on-chain.ts:305]: `displayCurrency` needs to be updated (similar to intraledger and lightning). Also `additionalCreditMetadata` needs to be updated

**Review by dolcalmi — CHANGES_REQUESTED** [review]
looks good, just minor changes

File: src/domain/shared/primitives.ts (hunk 1 above)
Comment (vindard) [reply src/domain/shared/primitives.ts:101]: Ah with the current implementation no, but I plan to change these to bigints and have a similar interface as `paymentAmountFromNumber` in a follow-up PR... can revert this change for now if we think we should

File: src/app/payments/send-intraledger.ts [hunk 5]
```diff
@@ -286,9 +286,9 @@ const executePaymentViaIntraledger = async <
         LedgerFacade.WalletIdIntraledgerLedgerMetadata({
           paymentAmounts: paymentFlow,
 
-          amountDisplayCurrency: converter.fromUsdAmount(paymentFlow.usdPaymentAmount),
+          amountDisplayCurrency: amountDisplayCurrencyAsNumber,
           feeDisplayCurrency: 0 as DisplayCurrencyBaseAmount,
-          displayCurrency: DisplayCurrency.Usd,
+          displayCurrency,
```
[the remark this answers is not in the document]
Comment (vindard) [reply src/app/payments/send-intraledger.ts:291]: Addressed in the very next PR #2384

File: src/app/payments/send-lightning.ts (hunk 3 above)
Comment (vindard) [reply src/app/payments/send-lightning.ts:434]: https://github.com/GaloyMoney/galoy/pull/2383#discussion_r1122315709

File: src/app/wallets/send-on-chain.ts (hunk 4 above)
Comment (vindard) [reply src/app/wallets/send-on-chain.ts:305]: https://github.com/GaloyMoney/galoy/pull/2383#discussion_r1122315709

File: src/app/payments/send-lightning.ts (hunk 2 above)
Comment (vindard) [reply src/app/payments/send-lightning.ts:342]: Done in next PR https://github.com/GaloyMoney/galoy/pull/2384/commits/ba909150b62c16f839845e282e9b25bd4577721a

**Review by dolcalmi — APPROVED** [review]

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**